File: Ordenacao1/cpp/src/main.cpp
#include <iostream>
#include <vector>

#include "selection_sort.h"
#include "insertion_sort.h"
#include "merge_sort.h"
#include "quick_sort.h"
#include "heap_sort.h"

#define READ_ALL while( std::cin >> aux ) { vec.push_back(aux); }

int main(int argc, const char *argv[]) {

  if(argc < 2) {
    std::cout << "Usage: bin/ord1 <ALGO> < input.file > output.file" << '\n';
    return 1;
  }

  std::vector<int> vec;
  int aux;

  switch(argv[1][0]) {
    case '1':
      READ_ALL;
      selection_sort(vec);
      break;

    case '2':
      READ_ALL;
      insertion_sort(vec);
      break;

    case '3':
      READ_ALL;
      vec = merge_sort(vec);
      break;

    case '4':
      READ_ALL;
      quick_sort(vec, 0, vec.size()-1);
      break;

    case '5':
      READ_ALL;
      heap::heap_sort(vec);
      break;

    default:
      std::cout << "Invalid algorithm" << '\n';
      break;
  }

  for(aux: vec) {
    std::cout << aux << ' ';
  }
  std::cout << std::endl;

  return 0;
}

File: Ordenacao1/rust/src/main.rs
use std::{io, env};

mod selection_sort;

fn main() {
    if let Some(arg1) = env::args().nth(1) {

        let mut vec : Vec<usize> = Vec::new();

        let mut input_text = String::new();
        'read_loop: loop {
            match io::stdin()
            .read_line(&mut input_text) {
                Ok(n) => {
                    if n == 0 {
                        break 'read_loop;
                    }
                },
                Err(..) => break 'read_loop,
            }

            let mut numbers: Vec<usize> = input_text
                .split_whitespace()
                .map(|s| s.parse().unwrap())
                .collect();
            vec.append(&mut numbers);
        }

        match arg1.as_ref() {
            "1" => {
                selection_sort::sort(&mut vec);
            },
            "2" => {
                selection_sort::sort(&mut vec);
            },
            _ => {
                println!("Invalid algorithm", );
            }
        }

        for num in vec {
            println!("{:?}", num);
        }

    } else {

        println!("Usage: ./ord <ALGO> < input.file > output.file", );

    }
}
